File: app/src/main/java/nl/sjtek/client/android/fragments/FragmentMusic.java
package nl.sjtek.client.android.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import nl.sjtek.client.android.R;
import nl.sjtek.client.android.api.API;
import nl.sjtek.client.android.api.Action;
import nl.sjtek.client.android.api.Arguments;

/**
 * Fragment for the Mopidy web interface.
 * Displays extra toolbar action buttons for controlling Mopidy.
 */
public class FragmentMusic extends BaseFragmentWeb {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        setHasOptionsMenu(true);
        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    protected String getUrl() {
        return "http://music.sjtek.nl/musicbox_webclient/index.html#home";
    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        inflater.inflate(R.menu.fragment_music, menu);
        super.onCreateOptionsMenu(menu, inflater);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.action_shuffle:
                API.action(getContext(), Action.Music.SHUFFLE);
                return true;
            case R.id.action_clear:
                API.action(getContext(), Action.Music.CLEAR);
                return true;
            case R.id.action_start:
                API.action(getContext(), Action.Music.START, new Arguments().setDefaultPlaylist(getContext()));
            default:
                return super.onOptionsItemSelected(item);
        }
    }
}

File: app/src/main/java/nl/sjtek/client/android/fragments/FragmentMusic.kt
package nl.sjtek.client.android.fragments

import android.os.Bundle
import android.view.*
import nl.sjtek.client.android.R
import nl.sjtek.client.android.api.API
import nl.sjtek.control.data.actions.Actions

/**
 * Fragment for the Mopidy web interface.
 * Displays extra toolbar action buttons for controlling Mopidy.
 */
class FragmentMusic : BaseFragmentWeb() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        setHasOptionsMenu(true)
        return super.onCreateView(inflater, container, savedInstanceState)
    }

    override fun getUrl(): String {
        return "http://music.sjtek.nl/musicbox_webclient/index.html#home"
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        inflater.inflate(R.menu.fragment_music, menu)
        super.onCreateOptionsMenu(menu, inflater)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.action_shuffle -> {
                API.action(context, Actions.music.shuffle())
                true
            }
            R.id.action_clear -> {
                API.action(context, Actions.music.clear())
                true
            }
            R.id.action_start -> {
                //TODO Add URI
                API.action(context, Actions.music.start(uri = ""))
                super.onOptionsItemSelected(item)
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
}
